kyaml/fn/runtime/container/container_test.go

```diff
@@ -0,0 +1,203 @@
+  name: foo
+`,
+			expectedArgs: []string{
+				"run",
+				"--rm",
+				"-i", "-a", "STDIN", "-a", "STDOUT", "-a", "STDERR",
+				"--network", "test-1",
+				"--user", "nobody",
+				"--security-opt=no-new-privileges",
+			},
+			instance: Filter{Image: "example.com:version", Network: "test-1"},
+		},
+
+		{
+			name: "storage_mounts",
+			functionConfig: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: foo
+`,
+			expectedArgs: []string{
+				"run",
+				"--rm",
+				"-i", "-a", "STDIN", "-a", "STDOUT", "-a", "STDERR",
+				"--network", "none",
+				"--user", "nobody",
+				"--security-opt=no-new-privileges",
+				"--mount", fmt.Sprintf("type=%s,src=%s,dst=%s:ro", "bind", "/mount/path", "/local/"),
+				"--mount", fmt.Sprintf("type=%s,src=%s,dst=%s:ro", "volume", "myvol", "/local/"),
+				"--mount", fmt.Sprintf("type=%s,src=%s,dst=%s:ro", "tmpfs", "", "/local/"),
+			},
+			instance: Filter{
+				Image: "example.com:version",
+				StorageMounts: []runtimeutil.StorageMount{
+					{MountType: "bind", Src: "/mount/path", DstPath: "/local/"},
+					{MountType: "volume", Src: "myvol", DstPath: "/local/"},
+					{MountType: "tmpfs", Src: "", DstPath: "/local/"},
+				},
+			},
+		},
+	}
+
+	for i := range tests {
+		tt := tests[i]
+		t.Run(tt.name, func(t *testing.T) {
+			cfg, err := yaml.Parse(tt.functionConfig)
+			if !assert.NoError(t, err) {
+				t.FailNow()
+			}
+			tt.instance.Exec.FunctionConfig = cfg
+
+			os.Setenv("KYAML_TEST", "FOO")
+			tt.instance.setupExec()
+
+			// configure expected env
+			for _, e := range os.Environ() {
+				// the process env
+				tt.expectedArgs = append(tt.expectedArgs, "-e", strings.Split(e, "=")[0])
+			}
+			tt.expectedArgs = append(tt.expectedArgs, tt.instance.Image)
+
+			if !assert.Equal(t, "docker", tt.instance.Exec.Path) {
+				t.FailNow()
+			}
+			if !assert.Equal(t, tt.expectedArgs, tt.instance.Exec.Args) {
+				t.FailNow()
+			}
+		})
+	}
+}
+
+func TestFilter_Filter(t *testing.T) {
+	cfg, err := yaml.Parse(`apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: foo
+`)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	input, err := (&kio.ByteReader{Reader: bytes.NewBufferString(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: deployment-foo
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: service-foo
+`)}).Read()
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	instance := Filter{}
+	instance.Exec.FunctionConfig = cfg
+	instance.Exec.Path = "sed"
+	instance.Exec.Args = []string{"s/Deployment/StatefulSet/g"}
+	output, err := instance.Filter(input)
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+
+	b := &bytes.Buffer{}
+	err = kio.ByteWriter{Writer: b, KeepReaderAnnotations: true}.Write(output)
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+
+	if !assert.Equal(t, `apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: deployment-foo
+  annotations:
+    config.kubernetes.io/index: '0'
+    config.kubernetes.io/path: 'statefulset_deployment-foo.yaml'
```

Comment:
ok, the comments make more sense now that i've read the tests. :)


very nice tests BTW